contracts/WKLAY.sol

```diff
@@ -39,9 +40,10 @@ contract WKLAY {
     }
 
     function withdraw(uint256 wad) public {
-        require(balanceOf[msg.sender] >= wad);
+        require(balanceOf[msg.sender] >= wad, "WKLAY: amount exceeds balance");
         balanceOf[msg.sender] -= wad;
-        msg.sender.transfer(wad);
+        (bool success, ) = msg.sender.call{value: wad}("");
```

Comment:
now that we're using call, should we test if reentrancy attack is successfully mitigated?

Reply:
Yes, one of the way is to change the state before we transfer. the state change for balance `balanceOf[msg.sender] -= wad;` is done before transfer. 
We can also inherit reentrancy guard contract